addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/PolicyDefinitionUnitTest.java

```diff
@@ -368,4 +369,133 @@ void shouldAlwaysIncludeIdWhenSerializingARule(int id) throws JsonProcessingExce
+        assertThat(rules.get(0).getName(), is(equalTo("PluginTestHelper")));
+        assertThat(rules.get(0).getStrength(), is(equalTo("insane")));
+        assertThat(rules.get(0).getThreshold(), is(equalTo("high")));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"TEST_TAG", "TEST_.*"})
+    void shouldExcludeIncludedRulesUsingAlertTags(String tagPattern) {
+        // Given
+        String yamlStr =
+                String.format(
+                        """
+                defaultStrength: low
+                defaultThreshold: medium
+                alertTags:
+                  include:
+                  - .*
+                  exclude:
+                  - %s
+                """,
+                        tagPattern);
+        AutomationProgress progress = new AutomationProgress();
+        Yaml yaml = new Yaml();
+        Object data = yaml.load(yamlStr);
+
+        // When
+        policyDefinition.parsePolicyDefinition(data, "test", progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        assertThat(policyDefinition.getEffectiveRules().isEmpty(), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldNotAddSameRuleTwice() {
+        // Given
+        String yamlStr =
+                """
+                defaultStrength: low
+                defaultThreshold: 'off'
+                rules:
+                - id: 50000
+                  name: rule1
+                  strength: insane
+                  threshold: high
+                alertTags:
+                  include:
+                    - TEST_TAG
+                  exclude: []
+                  strength: low
+                  threshold: medium
+                """;
+        AutomationProgress progress = new AutomationProgress();
+        Yaml yaml = new Yaml();
+        Object data = yaml.load(yamlStr);
+
+        // When
+        policyDefinition.parsePolicyDefinition(data, "test", progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        List<Rule> rules = policyDefinition.getEffectiveRules();
+        assertThat(rules.size(), is(equalTo(1)));
+        assertThat(rules.get(0).getId(), is(equalTo(50000)));
+        assertThat(rules.get(0).getName(), is(equalTo("PluginTestHelper")));
+        assertThat(rules.get(0).getStrength(), is(equalTo("insane")));
+        assertThat(rules.get(0).getThreshold(), is(equalTo("high")));
+    }
+
+    @Test
+    void shouldLoadPlansWithNullAlertTagFields() {
+        // Given
+        String yamlStr =
+                """
+                defaultStrength: low
+                defaultThreshold: medium
+                alertTags:
+                  include: null
+                  exclude: null
+                  strength: null
+                  threshold: null
```

Comment:
It worth adding tests for both enums with invalid values (e.g. `mediun`, this leads to (IMO) cryptic user error message `threshold is marked non-null but is null`).